lib/__tests__/matches-selector.ts

```diff
@@ -0,0 +1,123 @@
+/* eslint-env jest */
+
+import TestAssertions from '../helpers/test-assertions';
+
+describe('assert.dom(...).matchesSelector()', () => {
+  let assert;
+
+  beforeEach(() => {
+    assert = new TestAssertions();
+  });
+
+  describe('matchesSelector: success scenarios', () => {
+    test('succeeds for one element matching the selector', () => {
+      document.body.innerHTML = '<div><p class="first"></p><p></p><p class="last"></p></div>';
+
+      assert.dom('p.last').matchesSelector('div>p:last-child');
+
+      expect(assert.results).toEqual([{
+        actual: 1,
+        expected: 1,
+        message: 'The element selected by p.last also matches the selector div>p:last-child.',
+        result: true
+      }]);
+    });
+
+    test('succeeds for multiple elements, all sucessfully matched', () => {
+      document.body.innerHTML = '<div><p class="first"></p><p></p><p class="last"></p></div>.';
+
+      assert.dom('p').matchesSelector('div>p');
+
+      expect(assert.results).toEqual([{
+        actual: 3,
+        expected: 3,
+        message: '3 elements, selected by p, also match the selector div>p.',
+        result: true
+      }]);
+    });
+  });
```

Comment:
this is missing test cases where an `Element` is passed directly to `assert.dom(...)`. have a look at the other test suites for examples.

Reply:
So I did accomodate instances of Element passed to the assertion and looked at how hasClass does it.